test/suite/test_compress03.py

```diff
@@ -0,0 +1,113 @@
+
+    uri = "table:test_compress03"
+    nrows = 100000
+    valuea = "aaaaa"
+    valueb = "bbbbb"
+
+    def conn_config(self):
+        return f'statistics=(fast),statistics_log=(json,on_close,wait=1,sources=(\"file:\"))'
+
+    # Load the compression extension, skip the test if missing.
+    def conn_extensions(self, extlist):
+        extlist.skip_if_missing = True
+        extlist.extension('compressors', self.compressors)
+
+    def verify_stats(self, uri):
+        stat_cursor = self.session.open_cursor('statistics:' + uri, None, 'statistics=(fast)')
+        total_reads = stat_cursor[stat.dsrc.compress_read][2]
+        total_reads_hist = stat_cursor[stat.dsrc.compress_read_hist_ratio_2][2]
+        total_reads_hist += stat_cursor[stat.dsrc.compress_read_hist_ratio_4][2]
+        total_reads_hist += stat_cursor[stat.dsrc.compress_read_hist_ratio_8][2]
+        total_reads_hist += stat_cursor[stat.dsrc.compress_read_hist_ratio_16][2]
+        total_reads_hist += stat_cursor[stat.dsrc.compress_read_hist_ratio_32][2]
+        total_reads_hist += stat_cursor[stat.dsrc.compress_read_hist_ratio_64][2]
+        total_reads_hist += stat_cursor[stat.dsrc.compress_read_hist_ratio_max][2]
+        # We expect at least a read of compressed data.
+        assert total_reads and total_reads == total_reads_hist
+
+        total_writes = stat_cursor[stat.dsrc.compress_write][2]
+        total_writes_hist = stat_cursor[stat.dsrc.compress_write_hist_ratio_2][2]
+        total_writes_hist += stat_cursor[stat.dsrc.compress_write_hist_ratio_4][2]
+        total_writes_hist += stat_cursor[stat.dsrc.compress_write_hist_ratio_8][2]
+        total_writes_hist += stat_cursor[stat.dsrc.compress_write_hist_ratio_16][2]
+        total_writes_hist += stat_cursor[stat.dsrc.compress_write_hist_ratio_32][2]
+        total_writes_hist += stat_cursor[stat.dsrc.compress_write_hist_ratio_64][2]
+        total_writes_hist += stat_cursor[stat.dsrc.compress_write_hist_ratio_max][2]
+        # We expect at least a write of compressed data.
+        assert total_writes and total_writes == total_writes_hist
+
+        stat_cursor.close()
+
+    def large_updates(self, uri, value, ds, nrows, ts):
+        cursor = self.session.open_cursor(uri)
+        for i in range(0, nrows):
+            self.session.begin_transaction()
+            cursor[ds.key(i)] = value
+            self.session.commit_transaction('commit_timestamp=' + self.timestamp_str(ts))
+        cursor.close()
+
+    def test_compress03(self):
+        ds = SimpleDataSet(self, self.uri, 0, key_format="S", value_format="S", config=f"block_compressor={self.compressors}")
+        ds.populate()
+
+        self.large_updates(self.uri, self.valuea, ds, self.nrows, 1)
+        self.session.checkpoint()
+        self.large_updates(self.uri, self.valuea * 10, ds, self.nrows, 2)
+        self.session.checkpoint()
+        self.large_updates(self.uri, self.valuea * 100, ds, self.nrows, 3)
+        self.session.checkpoint()
```

Comment:
I feel like less variation in updates is probably required here. This test is doing quite a lot of work - hopefully more than is necessary to figure out if the stats are working.